ocrd/ocrd/resource_manager.py

```diff
@@ -55,16 +56,27 @@ def list_installed(self, executable=None):
         List installed resources, matching with registry by ``name``
         """
         ret = []
-        for executable in [executable] if executable else self.database.keys():
+        if executable:
+            all_executables = [executable]
+        else:
+            # resources we know about
+            all_executables = list(self.database.keys())
+            # resources in the file system
+            parent_dirs = [XDG_CACHE_HOME, XDG_CONFIG_HOME, XDG_DATA_HOME]
+            if 'VIRTUAL_ENV' in environ:
+                parent_dirs += [join(environ['VIRTUAL_ENV'], 'share')]
+            for parent_dir in parent_dirs:
+                all_executables += [x for x in listdir(parent_dir) if x.startswith('ocrd-')]
```

Comment:
This does cover all circumstances for unregistered resources I could think of, but it also includes false positives now:
```
ocrd-im6convert
- ocrd-tool.json (???)
  ???
```

It seems that `venv/share/ocrd-im6convert/ocrd-tool.json` matched the `startswith` pattern here. So require `os.path.isdir(x)`, too?

Reply:
Yeah, I saw that too. But `isdir(x)` wouldn't help because `ocrd-im6convert` is a directory. I think I'll just make a new im6convert release with `SHAREDIR=$VIRTUAL_ENV/share/ocrd_im6convert` (note the `_` instead of `-`).

Reply:
Yes, that makes sense. (ocrd_olena, ocrd_pagetopdf and ocrd_fileformat all use the underscore already.)